pkg/linkedlist/linkedlist.go

```diff
@@ -0,0 +1,105 @@
+// Author: Neel Shah, 2020
+// linkedlist.go contains the CRUD methods for the linked-list ADT
+// to be used as part of the hash table ADT.
+
+package linkedlist
+
+import (
+	"bytes"
+
+	"github.com/neelkshah/pandora/config"
+)
+
+// LinkedList is the data structure used for the hash table.
+type LinkedList struct {
+	head  *linkedListNode
+	tail  *linkedListNode
+	count int
+}
+
+// linkedListNode is the data structure that forms the LinkedList.
+type linkedListNode struct {
+	values   []valueNode
+	nextNode *linkedListNode
+}
+
+// valueNode contains a single value to be stored in the LinkedList.
+type valueNode struct {
+	key   []byte
```

Comment:
Why are you converting an integer key into a byte array?

Reply:
* I want to keep the linkedlist implementation uncoupled from the type discussion. 
* I'm using unsigned 64 bit ints (`uint64`) since these are essentially stored as byte arrays, as are strings. (And we don't need negatives)
* We will add a layer on top of this which will call the linkedlist API after converting the key/value values into byte arrays.
* The cost of the conversion is very low because of the existing underlying byte array representation.
* One outcome of this is that we aren't using a vacuous interface.